python-stdlib/typing/typing.py

```diff
@@ -0,0 +1,154 @@
+def cast(type, val):
+    return val
+
+
+def get_origin(type):
+    return None
+
+
+def get_args(type):
+    return ()
+
+
+def no_type_check(arg):
+    return arg
+
+
+def overload(func):
+    return None
+
+
+class _AnyCall:
+    def __init__(*args, **kwargs):
+        pass
+
+    def __call__(*args, **kwargs):
+        pass
+
+
+_anyCall = _AnyCall()
+
+
+class _SubscriptableType:
+    def __getitem__(self, arg):
+        return _anyCall
+
+
+_Subscriptable = _SubscriptableType()
+
+
+def TypeVar(type, *types):
+    return None
+
+
+def NewType(name, type):
+    return type
+
+
+class Any:
+    pass
+
+
+class BinaryIO:
+    pass
+
+
+class ClassVar:
+    pass
+
+
+class Final:
+    pass
+
+
+class Hashable:
+    pass
+
+
+class IO:
+    pass
+
+
+class NoReturn:
+    pass
+
+
+class Sized:
+    pass
+
+
+class SupportsInt:
+    pass
+
+
+class SupportsFloat:
+    pass
+
+
+class SupportsComplex:
+    pass
+
+
+class SupportsBytes:
+    pass
+
+
+class SupportsIndex:
+    pass
+
+
+class SupportsAbs:
+    pass
+
+
+class SupportsRound:
+    pass
+
+
+class TextIO:
+    pass
+
+
+AnyStr = str
+Text = str
+Pattern = str
+Match = str
+TypedDict = dict
+
+AbstractSet = _Subscriptable
+AsyncContextManager = _Subscriptable
+AsyncGenerator = _Subscriptable
+AsyncIterable = _Subscriptable
+AsyncIterator = _Subscriptable
+Awaitable = _Subscriptable
+Callable = _Subscriptable
+ChainMap = _Subscriptable
+Collection = _Subscriptable
+Container = _Subscriptable
+ContextManager = _Subscriptable
+Coroutine = _Subscriptable
+Counter = _Subscriptable
+DefaultDict = _Subscriptable
+Deque = _Subscriptable
+Dict = _Subscriptable
+FrozenSet = _Subscriptable
+Generator = _Subscriptable
+Generic = _Subscriptable
+Iterable = _Subscriptable
+Iterator = _Subscriptable
+List = _Subscriptable
+Literal = _Subscriptable
+Mapping = _Subscriptable
+MutableMapping = _Subscriptable
+MutableSequence = _Subscriptable
+MutableSet = _Subscriptable
+NamedTuple = _Subscriptable
+Optional = _Subscriptable
+OrderedDict = _Subscriptable
+Sequence = _Subscriptable
+Set = _Subscriptable
+Tuple = _Subscriptable
+Type = _Subscriptable
+Union = _Subscriptable
+
+TYPE_CHECKING = False
```

Comment:
```suggestion
from micropython import const

TYPE_CHECKING = const(False)


if TYPE_CHECKING:
    # Type Checkers on PC will "override" this dynamically to turn on TYPE_CHECKING and include below

    def TypeVar(type, *types):
        return None

    def NewType(name, type):
        return type

    def cast(type, val):
        return val

    def get_origin(type):
        return None

    def get_args(type):
        return ()

    def no_type_check(arg):
        return arg

    def overload(func):
        return None

    class _AnyCall:
        def __init__(*args, **kwargs):
            pass

        def __call__(*args, **kwargs):
            pass

    _anyCall = _AnyCall()

    class _SubscriptableType:
        def __getitem__(self, arg):
            return _anyCall

    _Subscriptable = _SubscriptableType()

    class Any:
        pass

    class BinaryIO:
        pass

    class ClassVar:
        pass

    class Final:
        pass

    class Hashable:
        pass

    class IO:
        pass

    class NoReturn:
        pass

    class Sized:
        pass

    class SupportsInt:
        pass

    class SupportsFloat:
        pass

    class SupportsComplex:
        pass

    class SupportsBytes:
        pass

    class SupportsIndex:
        pass

    class SupportsAbs:
        pass

    class SupportsRound:
        pass

    class TextIO:
        pass

    AnyStr = str
    Text = str
    Pattern = str
    Match = str
    TypedDict = dict

    AbstractSet = _Subscriptable
    AsyncContextManager = _Subscriptable
    AsyncGenerator = _Subscriptable
    AsyncIterable = _Subscriptable
    AsyncIterator = _Subscriptable
    Awaitable = _Subscriptable
    Callable = _Subscriptable
    ChainMap = _Subscriptable
    Collection = _Subscriptable
    Container = _Subscriptable
    ContextManager = _Subscriptable
    Coroutine = _Subscriptable
    Counter = _Subscriptable
    DefaultDict = _Subscriptable
    Deque = _Subscriptable
    Dict = _Subscriptable
    FrozenSet = _Subscriptable
    Generator = _Subscriptable
    Generic = _Subscriptable
    Iterable = _Subscriptable
    Iterator = _Subscriptable
    List = _Subscriptable
    Literal = _Subscriptable
    Mapping = _Subscriptable
    MutableMapping = _Subscriptable
    MutableSequence = _Subscriptable
    MutableSet = _Subscriptable
    NamedTuple = _Subscriptable
    Optional = _Subscriptable
    OrderedDict = _Subscriptable
    Sequence = _Subscriptable
    Set = _Subscriptable
    Tuple = _Subscriptable
    Type = _Subscriptable
    Union = _Subscriptable

else:
    # On-device when compiled to mpy the above should be stripped out leaving just this section.
    class _TypeIgnore:
        def __init__(*args, **kwargs):
            pass

        def __call__(self, *args, **kwargs):
            return self

        def __getitem__(self, attr):
            return self

    _type_ignore = _TypeIgnore()

    def __getattr__(attr):
        return _type_ignore
```

Reply:
Not sure how well it works suggesting this inline, but this is my attempt at providing an on-device simplified runtime stub, but keeping the "full" details for PC based type checking. `mpy-cross` does strip out the bulk of it, compiled to `mpy` beforehand was about 1.7KB, with this change it brings it down to 289 bytes for me.